pkg/server/cloud/common.go

```diff
@@ -138,5 +138,10 @@ func ToSaneVolumeSize(volSizeSpec string) int32 {
 		klog.Errorln("Empty volume size found in resource spec, setting to reasonable value")
 		volSizeGiB = 8
 	}
+	if image.VolumeDiskSize != nil {
```

Comment:
It's a minor issue: I would move this check ahead of the `if volSizeGiB == 0 {` check.  That check is a bit of a fail-safe to ensure users don't end up with a zero-sized volume. I also believe it still gets used in our functional tests.  The best  order is to first choose `*image.VolumeDiskSize` if it is available and, if not, to fall back to the arbitrarily chosen size of 8GB.  In other words: Do the thing that makes the most sense first and, if that fails, do something we only hope works.

Reply:
Good point, moved this if block :)